templates/CRM/Contribute/Form/Contribution/Main.extra.tpl

```diff
@@ -0,0 +1,46 @@
+{literal}
+  <script type="text/javascript">
+    CRM.$(function($) {
+      if (cj('#discountcode').length > 0) {
+        skipPaymentMethod();
+      }
+    });
+    function showHidePayment(flag) {
+      var payment_options = cj(".payment_options-group");
+      var payment_processor = cj("div.payment_processor-section");
+      var payment_information = cj("div#payment_information");
+      if (flag) {
+        payment_options.hide();
+        payment_processor.hide();
+        payment_information.hide();
+        // also unset selected payment methods
+        cj('input[name="payment_processor"]').removeProp('checked');
```

Comment:
According to the [jQuery docs](https://api.jquery.com/removeProp/) this can cause a fatal error - you should use `.prop('checked', false)` instead of `.removeProp('checked')`.